### PR TITLE
Dynamics365

### PR DESCRIPTION
**Pull request by greenc123**

### CODE REVIEW

File: server/getAccessToken.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+const params = new URLSearchParams();
+params.append('resource', 'https://fscpreassessmentdemo.crm.dynamics.com');
+params.append('client_id', '232bc5fc-3143-4783-abd3-eaed28124f15');
+params.append('grant_type', 'password');
+params.append('username', 'service@fscpreassessmentdemo.onmicrosoft.com');
+params.append('password', 'ZKHaXIzFwblIUgUXErcdrw7hu9');
+params.append('client_secret', 'WUJ6l9iu5I-vUCd689PKyl.BlP_h~tn2yo');
+
+const getAccessToken = async () => {
+  const url =
```
Comment (allopez7) [inline server/getAccessToken.js:13]: add try/catch block

File: server/getCountryData.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const fetch = require('node-fetch');
+const getAccessToken = require('./getAccessToken');
+
+const getCountryData = getAccessToken().then(token => {
+  const url =
```
Comment (allopez7) [inline server/getCountryData.js:5]: check if token is defined

File: server/getCountryData.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    'https://fscpreassessmentdemo.api.crm.dynamics.com/api/data/v9.1/fsc_countrydatas';
+
+  return fetch(url, {
```
Comment (allopez7) [inline server/getCountryData.js:8]: maybe change this function to async/await with try/catch as well.

Comment (greenc123) [reply server/getCountryData.js:8]: You're going to have to submit a PR. The way our code is structured I'm not sure how to remove the fetch.then setup. Async and await breaks it.

File: server/getCountryData.js [hunk 4]
```diff
@@ -0,0 +1,24 @@
+const fetch = require('node-fetch');
+const getAccessToken = require('./getAccessToken');
+
+const getCountryData = getAccessToken().then((token) => {
+  if (!token) return
+  const url =
+    'https://fscpreassessmentdemo.api.crm.dynamics.com/api/data/v9.1/fsc_countrydatas';
+
+  try {
+  fetch(url, {
```
Comment (allopez7) [inline server/getCountryData.js:10]: const countryData = await fetch....

File: server/getCountryData.js [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    headers: {
+      Accept: 'application/json',
+      Authorization: `Bearer ${token}`,
+    },
+  })
+  .then(res => res.json())
```
Comment (allopez7) [inline server/getCountryData.js:16]: const data = await response.json()

File: server/getCountryData.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  .then(data => console.log('data', data))
```
Comment (allopez7) [inline server/getCountryData.js:17]: return data, not entirely sure if there is a key here.

**Pull request merged**